include/circt/Dialect/Sim/SimOps.td

```diff
@@ -59,7 +59,7 @@ def FatalOp : SimOp<"fatal"> {
 
 def DPIFuncOp : SimOp<"func.dpi",
      [IsolatedFromAbove, Symbol, OpAsmOpInterface,
-      FunctionOpInterface, HasParent<"mlir::ModuleOp">]> {
+      FunctionOpInterface]> {
```

Comment:
This is similar change to https://github.com/llvm/circt/commit/09df7f53ed0dd264e77e7811f87dfab4d73e7f6e. It was necessary to put DPI func op into circuit op to make the lowering composable.